src/components/screens/DraggableGrid.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ReactElement, useContext } from 'react';
+import Animated, {
+  useAnimatedRef,
+  useAnimatedScrollHandler,
+  useSharedValue
+} from 'react-native-reanimated';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+import { device, normalize } from '../../config';
+import { OrientationContext } from '../../OrientationProvider';
+
+import { DraggableItem, Positions } from './DraggableItem';
+
+type Props = {
+  children: ReactElement<{ draggableId: string; key: string }>[];
+  onDragEnd: (diff: Positions) => void;
+};
+
+export const DraggableGrid = ({ children, onDragEnd }: Props) => {
+  const scrollY = useSharedValue(0);
+  const scrollView = useAnimatedRef<Animated.ScrollView>();
+  const positions = useSharedValue<Positions>(
+    Object.assign(
+      {},
+      ...children.map((child, index) => ({ [child.props.draggableId?.replace('​', '')]: index }))
+    )
+  );
+  const onScroll = useAnimatedScrollHandler({
+    onScroll: ({ contentOffset: { y } }) => {
+      scrollY.value = y;
+    }
```

Comment:
Can `x` be used with `y` here? Why did you choose only the `y` coordinate?

Reply:
i cannot answer that, it's from here https://github.com/wcandillon/can-it-be-done-in-react-native/blob/master/season4/src/Chrome/SortableList.tsx

Reply:
All right. 👍